src/hooks/common/use-base-url.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { useMemo } from 'react'
+import getConfig from 'next/config'
+import { useRouter } from 'next/router'
+
+export enum ORIGIN {
+  EDITOR,
+  RENDERER,
+  CURRENT_PAGE
+}
+
+/**
+ * Calculates the base url for relative URLs.
+ */
+export const useBaseUrl = (origin = ORIGIN.CURRENT_PAGE): string => {
```

Comment:
This will also be used for the backend, correct? But it only works because it's assumed that backend and frontend are running on the same url, right?